clippy_lints/src/single_letter_idents.rs

```diff
@@ -0,0 +1,52 @@
+use clippy_utils::{diagnostics::span_lint, source::snippet};
+use itertools::Itertools;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_lint::{EarlyContext, EarlyLintPass, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::symbol::Ident;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for idents which comprise of a single letter.
+    ///
+    /// Note: This lint can be very noisy when enabled; it even lints generics! it may be desirable
```

Comment:
It should probably not lint generics, as single-letter generics are the idiomatic form to put them.

Reply:
I tried this, but couldn't quite get it to work. Getting the `Res` from a `PathSegment` (field on `Node::TyParam`) doesn't work, but of course, that would be solved by just accessing the `res` field on it :sweat_smile: ... I'll probably add a configuration option for that later, along with fields